hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallServicePrecompileTest.java

```diff
@@ -68,26 +100,1043 @@ private static long getValue(SupportedContractModificationFunctions contractFunc
+        final var expectedTokenKeys = new ArrayList<PrecompileTestContract.TokenKey>();
+        expectedTokenKeys.add(new PrecompileTestContract.TokenKey(BigInteger.ONE, getKeyValue(tokenEntity.getKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(2), getKeyValue(token.getKycKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(4), getKeyValue(token.getFreezeKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(8), getKeyValue(token.getWipeKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(16), getKeyValue(token.getSupplyKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(32), getKeyValue(token.getFeeScheduleKey())));
+        expectedTokenKeys.add(
+                new PrecompileTestContract.TokenKey(BigInteger.valueOf(64), getKeyValue(token.getPauseKey())));
+
+        return expectedTokenKeys;
+    }
+
+    private String getAliasFromEntity(Entity entity) {
+        return Address.fromHexString(Bytes.wrap(entity.getEvmAddress()).toHexString())
+                .toHexString();
+    }
+
+    private String getAddressFromEntity(Entity entity) {
+        return EntityIdUtils.asHexedEvmAddress(new Id(entity.getShard(), entity.getRealm(), entity.getId()));
+    }
+
+    private String getAddressFromEntityId(EntityId entity) {
+        return EntityIdUtils.asHexedEvmAddress(new Id(entity.getShard(), entity.getRealm(), entity.getNum()));
+    }
+
+    private String getAddressFromEvmAddress(byte[] evmAddress) {
+        return Address.wrap(Bytes.wrap(evmAddress)).toHexString();
+    }
+
+    private KeyValue getKeyValue(byte[] serializedKey) {
+        try {
+            final var key = Key.parseFrom(serializedKey);
+            return new KeyValue(
+                    false,
+                    key.getContractID().hasContractNum()
+                            ? EntityIdUtils.asTypedEvmAddress(key.getContractID())
+                                    .toHexString()
+                            : Address.ZERO.toHexString(),
+                    key.getEd25519().toByteArray(),
+                    key.getECDSASecp256K1().toByteArray(),
+                    key.getDelegatableContractId().hasContractNum()
+                            ? EntityIdUtils.asTypedEvmAddress(key.getDelegatableContractId())
+                                    .toHexString()
+                            : Address.ZERO.toHexString());
+        } catch (InvalidProtocolBufferException e) {
+            throw new IllegalArgumentException("Unable to parse key", e);
+        }
+    }
+
+    private HederaToken convertTokenEntityToHederaToken(final Token token) {
+        final var entity = domainBuilder.entity().get();
```

Comment:
Believe these two domain classes should be linked.
```suggestion
        final var entity = domainBuilder.entity().customize(e -> e.id(token.getTokenId()).get();
```